hw/ip/hmac/doc/programmers_guide.md

```diff
@@ -77,6 +77,29 @@ Such attempts are discarded.
 The [`KEY`](registers.md#key) register ignores any attempt to access the secret key in the middle of the process.
 If the software tries to update the KEY, the IP reports an error through the Error FIFO. The error code is `SwUpdateSecretKeyInProcess`, `0x0003`.
 
+## Saving and restoring the context
+
+Software can let the HMAC IP process multiple message streams in a time-interleaved fashion by saving and restoring the context (i.e., parts of the hardware-internal state).
+
+Such context switches are possible only at the boundary of complete message blocks (512 bit for SHA-256).
+When SW doesn't know each instant at which a full message block is available, it can buffer data in memory until a block is full and only write HMAC's FIFOs once the buffer in memory contains a full message block.
+
+The context that needs to be saved and restored is in the following registers: `CFG`, `DIGEST_*`, and `MSG_LENGTH_*`.
```

Comment:
Would it make sense to add actual references to the register documentation here, i.e., `CFG -> [CFG](registers.md#cfg)`?

Reply:
Thanks, added